xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,165 @@
+from dataclasses import dataclass
+import argparse
+import difflib
+import json
+import subprocess
+import re
+import os
+from typing import List, Optional
+
+# Path to compiled C++ executable
+xml_converter: str = "../build/xml_converter"
+json_file_path: str = "../test_cases/test_expected_outputs.json"
+
+
+@dataclass
+class XMLConverter_arguments:
+    arg_input_xml: str = "--input-taco-path"
+    arg_output_xml: str = "--output-taco-path"
+    arg_input_proto: str = "--input-waypoint-path"
+    arg_output_proto: str = "--output-waypoint-path"
+    arg_split_proto: str = "--output-split-waypoint-path"
+
+
+def run_xml_converter(
+    input_xml: Optional[List[str]] = None,
+    output_xml: Optional[List[str]] = None,
+    input_proto: Optional[List[str]] = None,
+    output_proto: Optional[List[str]] = None,
+    split_output_proto: Optional[str] = None,
+) -> subprocess.CompletedProcess[str]:
+
+    # Build the command to execute the C++ program with the desired function and arguments
+    cmd: List[str] = [xml_converter]
+
+    if input_xml:
+        cmd += [XMLConverter_arguments.arg_input_xml] + input_xml
+    if output_xml:
+        cmd += [XMLConverter_arguments.arg_output_xml] + output_xml
+    if input_proto:
+        cmd += [XMLConverter_arguments.arg_input_proto] + input_proto
+    if output_proto:
+        cmd += [XMLConverter_arguments.arg_output_proto] + output_proto
+    if split_output_proto:
+        cmd += [XMLConverter_arguments.arg_split_proto] + [split_output_proto]
+
+    # Run the C++ program and capture its output
+    result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+    return result
```

Comment:
Wrapper functions like this I tend to return custom data instead of library return values. These functions are wrappers in the first place so returning a custom return type means that we can control both the input and the output function signatures and do any of the grunt work needed all within the wrapper function.
In this case I would return
* stdout
* stderr
* return code

Because those are the things that our particular program might care about outside of this function. You can return these as a tuple, or if you want to be extra, as a dataclass. Both options have pros and cons. Tuples can be implicitly deconstructed, dataclasses can handle more complex types but require a forward declaration and will only be used in a single place as one return value. If I were writing this I would return a tuple.